### PR TITLE
CMake docs & scripts

### PR DESCRIPTION
**Pull request by zetok**
fixed & rebased #4092 on top of master

---
This change is [Reviewable](https://reviewable.io/reviews/qtox/qtox/4097)

### CODE REVIEW

**Comment by zetok** [comment]
@Diadlo the only remaining part is getting qTox to finish compiling under osx with cmake, I have no idea what to do with this, care to take a look?

**Comment by Diadlo** [comment]
@zetok Seems, done :)

**Comment by Diadlo** [comment]
Reviewed 5 of 5 files at r2, 3 of 3 files at r3.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[INSTALL.md, line 70 at r3](https://reviewable.io:443/reviews/qtox/qtox/4097#-Kbamuryh5xOQPDPSucm:-Kbamuryh5xOQPDPSucn:btcewz0) ([raw file](https://github.com/qtox/qtox/blob/1a3345ef05e3158a25bcf6f3d0db7e0b946368c9/INSTALL.md#L70)):*
> ```Markdown
> | [libX11]        | >= 1.6.0 |
> 
> Disabled if dependencies are missing during compilation.
> ```

What about add switch or add `TODO`?

---

*[.travis/build-osx.sh, line 3 at r3](https://reviewable.io:443/reviews/qtox/qtox/4097#-KbanzKuWKAv7uppVXhO:-KbanzKvkUv5YYQABJnW:b-jedf4n) ([raw file](https://github.com/qtox/qtox/blob/1a3345ef05e3158a25bcf6f3d0db7e0b946368c9/.travis/build-osx.sh#L3)):*
> ```Shell
> #!/bin/bash
> #
> #    Copyright © 2017 by The qTox Project Contributors
> ```

2016-2017

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/qtox/qtox/4097)*

**Comment by zetok** [comment]
Review status: 7 of 8 files reviewed at latest revision, 2 unresolved discussions.

---

*[INSTALL.md, line 70 at r3](https://reviewable.io:443/reviews/qtox/qtox/4097#-Kbamuryh5xOQPDPSucm:-KbapvQ_X-AF83zQrTYd:b-dge3b) ([raw file](https://github.com/qtox/qtox/blob/1a3345ef05e3158a25bcf6f3d0db7e0b946368c9/INSTALL.md#L70)):*
<details><summary><i>Previously, Diadlo (Polshakov Dmitry) wrote…</i></summary><blockquote>

What about add switch or add `TODO`?
</blockquote></details>

I've added that at the bottom of the INSTALL.md, should it be moved somewhere else?

---

*[.travis/build-osx.sh, line 3 at r3](https://reviewable.io:443/reviews/qtox/qtox/4097#-KbanzKuWKAv7uppVXhO:-Kbapn24xDdbWwNOjoYV:b-896fix) ([raw file](https://github.com/qtox/qtox/blob/1a3345ef05e3158a25bcf6f3d0db7e0b946368c9/.travis/build-osx.sh#L3)):*
<details><summary><i>Previously, Diadlo (Polshakov Dmitry) wrote…</i></summary><blockquote>

2016-2017
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/qtox/qtox/4097)*

**Comment by Diadlo** [comment]
:lgtm_strong:

---

Reviewed 1 of 1 files at r4.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[INSTALL.md, line 70 at r3](https://reviewable.io:443/reviews/qtox/qtox/4097#-Kbamuryh5xOQPDPSucm:-KbaqM_GNi2nCDYd9hxh:bb710yj) ([raw file](https://github.com/qtox/qtox/blob/1a3345ef05e3158a25bcf6f3d0db7e0b946368c9/INSTALL.md#L70)):*
<details><summary><i>Previously, zetok wrote…</i></summary><blockquote>

I've added that at the bottom of the INSTALL.md, should it be moved somewhere else?
</blockquote></details>

Nope

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/qtox/qtox/4097#-:-KbaqQq9HkFI-wBk7XsZ:b-jbsf8z)*

**Comment by sudden6** [comment]
Reviewed 3 of 5 files at r2, 3 of 3 files at r3, 1 of 1 files at r4.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[INSTALL.md, line 742 at r4](https://reviewable.io:443/reviews/qtox/qtox/4097#-KbatgMGTom7tWomNhZx:-KbatgMHFtOvG7fFHr7p:b-38awnf) ([raw file](https://github.com/qtox/qtox/blob/32106372d1ffb67d654d9674c9536888da562574/INSTALL.md#L742)):*
> ```Markdown
> 
> - `SMILEYS`, values:
>   - if not defined or there is not supported value, all emoticon packs are
> ```

`if not defined or an unsupported value is passed, all emoticon packs are included`

---

*[INSTALL.md, line 744 at r4](https://reviewable.io:443/reviews/qtox/qtox/4097#-Kbau5-lIv16vX2hCYvW:-Kbau5-mSK-LjYsL6Rkh:b-14toln) ([raw file](https://github.com/qtox/qtox/blob/32106372d1ffb67d654d9674c9536888da562574/INSTALL.md#L744)):*
> ```Markdown
>   - if not defined or there is not supported value, all emoticon packs are
>     included
>   - `DISABLED` – disables support for emoticons
> ```

Support for emoticons is not disabled, just none are included. I suggest rewording to `don't include any emoticon packs, custom ones are still loaded`.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/qtox/qtox/4097)*

**Comment by zetok** [comment]
Review status: 7 of 8 files reviewed at latest revision, 3 unresolved discussions.

---

*[INSTALL.md, line 742 at r4](https://reviewable.io:443/reviews/qtox/qtox/4097#-KbatgMGTom7tWomNhZx:-Kbavp0T0PKdDDNVezeT:b-896fix) ([raw file](https://github.com/qtox/qtox/blob/32106372d1ffb67d654d9674c9536888da562574/INSTALL.md#L742)):*
<details><summary><i>Previously, sudden6 wrote…</i></summary><blockquote>

`if not defined or an unsupported value is passed, all emoticon packs are included`
</blockquote></details>

Done.

---

*[INSTALL.md, line 744 at r4](https://reviewable.io:443/reviews/qtox/qtox/4097#-Kbau5-lIv16vX2hCYvW:-Kbaw8u_lKufGtX33Atd:b-896fix) ([raw file](https://github.com/qtox/qtox/blob/32106372d1ffb67d654d9674c9536888da562574/INSTALL.md#L744)):*
<details><summary><i>Previously, sudden6 wrote…</i></summary><blockquote>

Support for emoticons is not disabled, just none are included. I suggest rewording to `don't include any emoticon packs, custom ones are still loaded`.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/qtox/qtox/4097)*

**Comment by sudden6** [comment]
:lgtm_strong:

---

Review status: 7 of 8 files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/qtox/qtox/4097#-:-KbaxDihOzOJX9hfkXvb:b-jbsf8z)*
